src/Mod/Part/Gui/TaskCheckGeometry.cpp

```diff
@@ -433,11 +433,22 @@ void TaskCheckGeometryResults::goCheck()
     for(const auto &sel :  selection) {
         selectedCount++;
         TopoDS_Shape shape = Part::Feature::getShape(sel.pObject,sel.SubName,true);
-        if (shape.IsNull())
+        if (shape.IsNull()) {
+            Base::Console().Warning(tr("CheckGeometry: Skipping %1, has null or no shape.")
+                                    .arg(QString::fromUtf8(sel.pObject->getNameInDocument())).toStdString().c_str());
+            Base::Console().Message("\n");
             continue;
+        }
+        if (Part::TopoShape(shape).getBoundBox().CalcDiagonalLength() > 1e+100) {
+            Base::Console().Warning(tr("CheckGeometry: Skipping %1, too large. (Infinite object?)")
+                                    .arg(QString::fromUtf8(sel.pObject->getNameInDocument())).toStdString().c_str());
+            Base::Console().Message("\n");
+            continue;
+        }
```

Comment:
Do we want warning messages for the shapes we skip, or should it silently continue as it did before and still does for `!currentSeparator`? (I didn't see any explanation in the description of this PR adding a warning)

Reply:
It is in the description, but I will remove the warnings.